web/pgadmin/tools/schema_diff/static/js/components/SchemaDiffCompare.jsx

```diff
@@ -140,9 +140,15 @@ export function SchemaDiffCompare({ params }) {
   }, []);
 
   useEffect(()=>{
-    let oldSchemaDiffData1 = getToolData(params.params?.toolDataId);
-    setOldSchemaDiffData(oldSchemaDiffData1);
-  },[]);
+    if(params.restore == 'true'){
+      async function fetchData() {
+        const response = await getToolContent(params.transId);
+        setOldSchemaDiffData(response?.data);
```

Comment:
setOldSchemaDiffData?

Reply:
This is used while restoring servers/dbs/schema information selected previously by the user .